acceptance-tests/domain_fronting_test.go

```diff
@@ -227,3 +215,33 @@ func buildRequest(address string, hostHeader string) *http.Request {
 	req.Host = hostHeader
 	return req
 }
+
+func expect400BadRequestNoHostHeader(haproxyIP string, tlsConfig *tls.Config) {
```

Comment:
I guess this is fine for now as we have only one use case. Maybe later we can turn this into something like a `tlsClient` that is usable for generic L4 requests. Similar to the httpClients we have now.

We could then pass the client with an expected response to the `expect` function and make it more generic.

Reply:
Unfortunately there isn't a tls client in the standard library -- you use a the config to directly create the connection, so we'd have to create our own.

Reply:
yes, I meant that. We would have to write our own little tlsClient somewhere. But let's do it only when needed.